src/Test/ResetDatabase.php

```diff
@@ -34,7 +33,9 @@ public static function _resetDatabase(): void
         $kernel = static::createKernel();
         $kernel->boot();
 
-        DatabaseResetter::resetDatabase($kernel);
+        if (self::shouldReset($kernel)) {
+            DatabaseResetter::resetDatabase($kernel);
+        }
```

Comment:
I've literally spent hours on this and I figured out that the kernel should not be booted before calling `StaticDriver::setKeepStaticConnections(false);`. 

Since the `shouldReset()` must access to the config, we need the kernel to be booted before calling it.

At first I was trying to do something like this.
```php
        $kernel = static::createKernel();
        $kernel->boot();

        if (!self::shouldReset($kernel)) {
            // $kernel->shutdown()
            return;
        }

        if ($isDAMADoctrineTestBundleEnabled = DatabaseResetter::isDAMADoctrineTestBundleEnabled()) {
            // disable static connections for this operation
            StaticDriver::setKeepStaticConnections(false);
        }

        DatabaseResetter::resetDatabase($kernel);
```
and all my tests resulted in errors such as:
> PDOException: SQLSTATE[42000]: Syntax error or access violation: 1305 SAVEPOINT DOCTRINE2_SAVEPOINT_4 does not exist